hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java

```diff
@@ -201,6 +201,10 @@ public void run() {
                 if (completionListener != null) {
                     completionListener.onException(e);
                 }
+            } catch (Throwable t) {
```

Comment:
@sancar This error is not directly dispatched to user thread but passed to `CompletionListener`. So what about throwing it if it is `OutOfMemoryError` after `CompletionListener` is notified. WDYT?